src/components/MessageList/MessageListInner.js

```diff
@@ -49,7 +50,21 @@ const insertDates = (messages) => {
       prevMessageDate = messages[i - 1].created_at.toDateString();
     }
 
-    if (i === 0 || messageDate !== prevMessageDate) {
+    if (!unread) {
+      unread = lastRead.getTime() < message.created_at.getTime();
```

Comment:
is `lastRead` always defined?

Reply:
It looks like it isn't: https://github.com/GetStream/stream-chat-js/blob/ca4f5a587d7d10c6a424b82eba0a359605f0023a/src/channel.ts#L1110

```suggestion
      unread = lastRead && lastRead.getTime() < message.created_at.getTime();
```

Reply:
logically it should not be undefined at this stage, but won't hurt to have an extra check 👍

Reply:
And will make TS less angry in the future x)